materialize-snowflake/sqlgen.go

```diff
@@ -0,0 +1,218 @@
+package main
+import (
+  "fmt"
+  "strings"
+
+	"encoding/json"
+  sql "github.com/estuary/connectors/materialize-sql"
+	"text/template"
+	"github.com/google/uuid"
+  "github.com/estuary/flow/go/protocols/fdb/tuple"
+)
+
+var jsonConverter sql.ElementConverter = func(te tuple.TupleElement) (interface{}, error) {
+  switch ii := te.(type) {
+  case []byte:
+    return json.RawMessage(ii), nil
+  case json.RawMessage:
+    return ii, nil
+  case nil:
+    return json.RawMessage(nil), nil
+  default:
+    return nil, fmt.Errorf("invalid type %#v for variant", te)
+  }
+}
+
+// snowflakeDialect returns a representation of the Snowflake SQL dialect.
+var snowflakeDialect = func () sql.Dialect {
+  var variantMapper = sql.NewStaticMapper("VARIANT", sql.WithElementConverter(jsonConverter))
+  var typeMappings = sql.ProjectionTypeMapper{
+    sql.ARRAY:   variantMapper,
+    sql.BINARY:  sql.NewStaticMapper("BINARY"),
+    sql.BOOLEAN: sql.NewStaticMapper("BOOLEAN"),
+    sql.INTEGER: sql.NewStaticMapper("INTEGER"),
+    sql.NUMBER:  sql.NewStaticMapper("DOUBLE"),
+    sql.OBJECT:  variantMapper,
+    sql.STRING: sql.StringTypeMapper{
+      Fallback: sql.NewStaticMapper("TEXT"),
```

Comment:
This used to be `STRING` and `TIMESTAMP` below. Is there a reason for them to change?

Reply:
I'm glad you noticed that, that's just copypasta error, thanks!